resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/docker/SparkDockerImageBuilder.scala

```diff
@@ -76,21 +78,34 @@ private[spark] class SparkDockerImageBuilder
     val pythonExec = sys.env.get("PYSPARK_DRIVER_PYTHON")
       .orElse(sys.env.get("PYSPARK_PYTHON"))
       .getOrElse("/usr/bin/python")
-    val builder = new ProcessBuilder(
+    val python_builder = new ProcessBuilder(
       Seq(pythonExec, "setup.py", "sdist").asJava)
-    builder.directory(new File(DOCKER_BUILD_PATH.toFile, "python"))
-    builder.redirectErrorStream(true) // Ugly but needed for stdout and stderr to synchronize
-    val process = builder.start()
+    python_builder.directory(new File(DOCKER_BUILD_PATH.toFile, "python"))
+    python_builder.redirectErrorStream(true) // Ugly but needed for stdout and stderr to synchronize
+    val process = python_builder.start()
     new RedirectThread(process.getInputStream, System.out, "redirect output").start()
-    val exitCode = process.waitFor()
-    if (exitCode != 0) {
-      logInfo(s"exitCode: $exitCode")
+    val exitCodePython = process.waitFor()
```

Comment:
I've always been wondering - would it be possible to do this bootstrapping from Maven and not in the test code? Seems like this should be an environment step. Docker images should theoretically be built at the Maven step as well but we know that this is harder to do.

Reply:
It technically is possible, but as we discussed for PySpark. I wasn't able to figure that out. Any recommendations would be helpful.